### PR TITLE
Disable VerifyFloatToUInt and VerifyFloatToNullableUInt tests

### PR DESCRIPTION
**Pull request by EgorBo**
Disables two tests, see https://github.com/dotnet/runtime/issues/47374.

Minimal repro for MSVC x86: https://github.com/dotnet/runtime/issues/47374#issuecomment-766348300

Godbolt with other compilers: https://godbolt.org/z/W74hTh

### CODE REVIEW

**Comment by EgorBo** [comment]
Ok I was wrong, the real reason is `(unsigned)floatValue` on MSVC compiled for x86_32 behaves differently than all other platforms.

File: src/coreclr/jit/compiler.hpp
Comment (tannergooding) [inline src/coreclr/jit/compiler.hpp:67]: Do we have tests covering these scenarios to ensure they are consistent across both current and future compilers?

Comment (tannergooding) [reply src/coreclr/jit/compiler.hpp:67]: Is the same not needed for `float`?

**Comment by tannergooding** [comment]
Should we log a bug on MSVC to see if this is something they want to fix in a future update, or if they need to keep the behavior for backcompat?

File: src/coreclr/jit/compiler.hpp
Comment (saucecontrol) [reply src/coreclr/jit/compiler.hpp:67]: float->uint casts are forced through double on x86

https://github.com/dotnet/runtime/blob/fdc8148e9a8626da75f98f462f814946af35e5f1/src/coreclr/jit/morph.cpp#L175-L180

Comment (tannergooding) [reply src/coreclr/jit/compiler.hpp:67]: Ah I see. I wonder if that's holdover from when we still used the x87 FPU stack

**Comment by SingleAccretion** [comment]
After giving this some more thought, I am wondering if the best long-term solution would be to just disable folding for constants that invoke implementation-specified behavior. This should a few birds with one stone:
1) We'll get cross-compilation consistency "for free". Today, as far as I can see, cross-compiling from `x64` to `arm64` will result in differences between the folding logic and the runtime logic (this is same issue that Roslyn had).
2) We'll be shielded from compilers changing the unspecified behavior.
3) We'll not have to worry about harmonizing the codegen with the folding logic and ensuring they produce the same results (no need for maintaining tables of `(Type <- Type)(Arch) -> results` combinations and similar).

This is already the behavior for non-finite constants. @tannergooding what do you think?

(To be clear, this is for a future PR, this fix to unblock CI looks good modulo the fact that I think we should check for `d <= TheLargestValueThatDoesNotOverflow`, or even just `d <= (double)uint.MaxValue`, to be on the safe side. E. g. `(unsigned)3.40282e+35` also exhibits this behavior on MSVC x86 as per my testing).

**Comment by EgorBo** [comment]
@tannergooding my goal was to unblock the CI asap and fix those two failing tests.

I was planning to add more test cases separately (e.g. these https://github.com/dotnet/roslyn/issues/37527) to list all the edge cases/differences

**Comment by tannergooding** [comment]
@EgorBo, I think it's likely better to disable the two tests right now and work on getting a fix up separately.

As @SingleAccretion mentioned, this impacts more than just the value being handled in the PR and it would likely be good to get some validation tests up simultaneously.

**Comment by tannergooding** [comment]
> @tannergooding what do you think?

I think having a temporary fix, such as deciding to not fold (or at least not fold when in R2R scenarios) is fine. But I think, long term, we would be in a better place by having these cases handled and well known, just like Roslyn did.

> no need for maintaining tables of (Type <- Type)(Arch) -> results combinations and similar

I'm not super concerned with this. We will already need the first half of the table to know which values are edge cases and it would be good (IMO) to have some minor documentation on what values users can expect for these values and therefore how they will differ.

> This is already the behavior for non-finite constants

This is a drawback, IMO. At least for `floating-point`, the IEEE 754 spec allows a certain amount of flexibility with handling of things like `NaN` without it being considered a "value changing optimization".
We aren't taking advantage of this and likely aren't generating the best code in certain scenarios. It's likely an edge case overall, but its still something most compilers handle.

**Comment by EgorBo** [comment]
@tannergooding makes sense, I've disabled the tests.

**Review by tannergooding — APPROVED** [review]

**Comment by benaadams** [comment]
@EgorBo might want to take Fixes out of summary so it doesn't close the other issue (or reopen after merge)

**Comment by hoyosjs** [comment]
Merging to unblock CI.

**Comment by akoeplinger** [comment]
Can we backport this to the release/6.0-preview1 branch? I saw these tests failing there too.

**Pull request merged**
